third_party/move/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -1141,8 +1147,13 @@ pub fn build_and_report_v2_driver(options: move_compiler_v2::Options) -> Compile
 pub fn build_and_report_no_exit_v2_driver(
     options: move_compiler_v2::Options,
 ) -> CompilerDriverResult {
-    let mut stderr = StandardStream::stderr(ColorChoice::Auto);
-    let mut emitter = options.error_emitter(&mut stderr);
+    let mut writer: Box<dyn WriteColor> = if options.print_errors {
```

Comment:
how about moving it into `error_emitter` because we can check the value of print_errors there?

Reply:
I had to move the `if` block to a separate function to avoid `options.error_writer()` to return local data. Still much nicer I think.